Writexlsx params

jmbarbone opened the pull request:
Resolves #192 and #187.

Adds the `buildWorkbook()` function to extract the relevant processes from `write.xlsx()`.  The method in which I used should also allow for all applicable parameters passed through `...` into the other function calls.  

I have also made some formatting changes and small improvements in some of the code I was looking at. A large portion is simplifying the use of `grep()` and `grepl()` and adding some `assert_*()` functions which should make parameter checking easier and reduce some of the duplicate code.

It may be worth while, too, to update some of the tests that use `write.xlsx(x, file); wb <- read.xlsx(file)` as `buildWorkbook()` now accomplishes that.

jmbarbone committed:
simplifies write.xlsx

jmbarbone committed:
extracts write.xlsx to buildWorkbook

still a WIP

jmbarbone committed:
Merge branch 'master' into writexlsx-params

jmbarbone committed:
formatting

jmbarbone committed:
simplifies options

jmbarbone committed:
adds argument for row.names

jmbarbone committed:
recreations names if rownames in TRUE

* when rownames is TRUE the first column (which already has * a name made) is removed
this makes sure that the made names start with X1, X2 rather than X2, X3

jmbarbone committed:
corrects loop

jmbarbone committed:
formatting

jmbarbone committed:
updates tests

jmbarbone committed:
adds asserts and updates class/inherit checks

jmbarbone committed:
adds min/max width options

jmbarbone committed:
adds temp_xlsx() utilf un

jmbarbone committed:
passes sep.names fromr eadWorkbook to read.xlsx

jmbarbone committed:
formatting

* simplifies x[grepl()] to grep(value = TRUE) when appropriate
* uses inherits() or is.*() for class checking
* formatting

jmbarbone committed:
simplifies grepls

speed is virtually the sample but grep(value = TRUE) less memory allocation than x[grepl()]

jmbarbone committed:
adds tests for tableName in buildWorkbook()

resolves #187

jmbarbone committed:
updates NEWS

jmbarbone committed:
row.names >> rowNames

jmbarbone committed:
corrects strings optiosn

JanMarvin commented on R/readWorkbook.R:
```diff
@@ -69,43 +69,47 @@
 #' }
```
why did you modify this file?

jmbarbone replied:
Mostly from adding `assert_*()`s and while testing for any differences in changing up `write.xlsx()` as there were a lot of testing code that used the `write.xlsx(x, file); read.xlsx(file)` process

JanMarvin commented on R/wrappers.R:
```diff
@@ -29,40 +29,40 @@
 #'   category = "something"
```
same for this file. only whitespace changes?

jmbarbone replied:
Some changes at  386.  Not entirely sure what all the white space changes are.  I'm using Rstudio `1.4.1103`, could there be some line return formatting causing this?

JanMarvin replied:
Rstudio has some formating options, removing trailing whitespace options, ensure empty line at the end of the file. For personal projects I usually stick with them, but all this unrelated noise make this pull request harder to read.
I'm not against the changes and from a quick glance they seem to improve the code, I just don't want to review all the lines of unrelated changes looking for possible issues in the new and changed code 😄

jmbarbone replied:
> Rstudio has some formating options, removing trailing whitespace options, ensure empty line at the end of the file. For personal projects I usually stick with them, but all this unrelated noise make this pull request harder to read.

Agreed.  I feel like this isn't going to be an isolated thing.  Temping now to make one big PR of just white-noise?

JanMarvin replied:
First of, that's not my call to make (poking @ycphs) and ideally it would be a decision unrelated to an IDE configuration.
Personally, I would prefer to introduce such changes only when applying code changes. Iirc correctly, such whitespace related bulk commits make git bisecting and blaming harder, but _I didn't check that_.
Finally, I think that this should not be a blocker for this pull request, but I won't volunteer right now to review it. (And I know projects, where such pull request would be closed without any hesitation.)

ycphs replied:
I totally agree with @JanMarvin. It is hard to read, but I will merge the PR. The next time it would be great if only the rows with actual changes is pushed.

JanMarvin commented on R/writeData.R:
```diff
@@ -14,6 +14,7 @@
 #' \code{c(startCol, startRow)}.
 #' @param colNames If \code{TRUE}, column names of x are written.
 #' @param rowNames If \code{TRUE}, data.frame row names of x are written.
+#' @param row.names Deprecated, please use \code{rowNames} instead
```
I do not get this. This looks as if you added and deprecated it in the very same moment?

jmbarbone replied:
`row.names` had been originally passed from `write.xlsx()` and converted to `rowNames` before `writeData()` and `writeDataTable()`.  This was also done with `col.names` and `colNames` but I missed that because the latter wasn't used in testing.

JanMarvin replied:
I understand. I initially was wondering, why you'd deprecate functions from a stable API, but using similar arguments looks like an improvement. 👍

JanMarvin commented on R/writeDataTable.R:
```diff
@@ -12,6 +12,7 @@
 #' A vector of the form c(startCol, startRow)
 #' @param colNames If \code{TRUE}, column names of x are written.
 #' @param rowNames If \code{TRUE}, row names of x are written.
+#' @param row.names Deprecated, please use \code{rowNames} instead
```
Same for this

jmbarbone replied:
As above.

JanMarvin reviewed (commented):
Coding looks nice and I like the wrappers for some of the functions, but I'm not gonna review a 3.5k diff of whitespace changes :smile:

jmbarbone commented on R/wrappers.R:
```diff
@@ -383,91 +383,95 @@ addWorksheet <- function(
   options("OutDec" = ".")
   on.exit(expr = options("OutDec" = od), add = TRUE)
-  
+
```
@JanMarvin some changes here

jmbarbone committed:
visible returns Workbook and adds examples

buildWorkbook() is intended to be used with an assignment operator, so an explicit return should reinforce this

jmbarbone committed:
adds warning for col.names

adds warning and tests for use of col.names
